taca/deliver/deliver.py

```diff
@@ -0,0 +1,385 @@
+        """ Fetch the database sample entries representing the instance's project
+            :returns: a list of json-formatted database sample entries
+            :raises DelivererDatabaseError: 
+                if an error occurred when communicating with the database
+        """
+        return self.wrap_database_query(
+            self.dbcon().project_get_samples,self.projectid)
+
+    @memoized
+    def sample_entry(self):
+        """ Fetch a database entry representing the instance's project and sample
+            :returns: a json-formatted database entry
+            :raises DelivererDatabaseError: 
+                if an error occurred when communicating with the database
+        """
+        return self.wrap_database_query(
+            self.dbcon().sample_get,self.projectid,self.sampleid)
+
+    def update_delivery_status(self, status="DELIVERED"):
+        """ Update the delivery_status field in the database to the supplied 
+            status for the project and sample specified by this instance
+            :returns: the result from the underlying api call
+            :raises DelivererDatabaseError: 
+                if an error occurred when communicating with the database
+        """
+        update_fn = self.dbcon().sample_update \
+            if self.sampleid is not None \
+            else self.dbcon().project_update
+        args = [self.projectid]
+        if self.sampleid is not None:
+            args.append(self.sampleid)
+        kwargs = {'delivery_status': status}
+        return self.wrap_database_query(
+            update_fn,
+            *args,
+            **kwargs)
+            
+    def wrap_database_query(self,query_fn,*query_args,**query_kwargs):
+        """ Wrapper calling the supplied method with the supplied arguments
+            :param query_fn: function reference in the CharonSession class that
+                will be called
+            :returns: the result of the function call
+            :raises DelivererDatabaseError: 
+                if an error occurred when communicating with the database
+        """
+        try:
+            return query_fn(*query_args,**query_kwargs)
+        except db.CharonError as ce:
+            raise DelivererDatabaseError(ce.message)
+            
+    def gather_files(self):
+        """ This method will locate files matching the patterns specified in 
+            the config and compute the checksum and construct the staging path
+            according to the config.
+            
+            The config should contain the key 'files_to_deliver', which should
+            be a list of tuples with source path patterns and destination path
+            patterns. The source path can be a file glob and can refer to a 
+            folder or file. File globs will be expanded and folders will be
+            traversed to include everything beneath.
+             
+            :returns: A generator of tuples with source path, 
```

Comment:
this is really nice @b97pla ! And very good choice with the generator, this will avoid really long lists of files from filling up memory and/or slowing down TACA